File: 프로그래머스/Lv1/명예의 전당.py
def solution(k, score):
    answer = []
    result = []
    
    for i in score:
        answer.append(i)
        if len(answer) > k:
            answer.remove(min(answer))
            
        result.append(min(answer))
        
    return result

print(solution(3, [10, 100, 20, 150, 1, 100, 200]))
print(solution(4, [0, 300, 40, 300, 20, 70, 150, 50, 500, 1000]))

File: 프로그래머스/Lv1/명예의 전당.js
/**
 * 내 풀이
 * @param {number} k 
 * @param {Array} score 
 * @returns 
 */
function solution(k, score) {
    var answer = [];
    var result = [];
    
    for(var i=0; i<score.length; i++){
        if(answer.length < k){
            answer.push(score[i]);
        }else if(score[i] > Math.min(...answer)){
            var idx = answer.indexOf(Math.min(...answer));
            answer[idx] = score[i];
        }
        
        result.push(Math.min(...answer));
    }
    return result;
}

/**
 * 다른사람 풀이
 * @param {number} k 
 * @param {Array} score 
 * @returns 
 */
function solution2(k, score) {
    var answer = [];

    return score.reduce((a, c) => {
        answer.push(c);
        answer.sort((a, b) => a - b);

        if(answer.length > k){
            answer.shift();
        }

        a.push(answer[0]);
        return a;
    }, [])
}

console.log(solution(3, [10, 100, 20, 150, 1, 100, 200]));
console.log(solution(4, [0, 300, 40, 300, 20, 70, 150, 50, 500, 1000]));